src/transformers/models/aria/modeling_aria.py

```diff
@@ -991,7 +991,7 @@ def _update_causal_mask(
         # order to dispatch on Flash Attention 2. This feature is not compatible with static cache, as SDPA will fail
         # to infer the attention mask.
         past_seen_tokens = past_key_values.get_seq_length() if past_key_values is not None else 0
-        using_static_cache = isinstance(past_key_values, StaticCache)
+        using_static_cache = past_key_values.is_compileable
```

Comment:
we want to use the fixed-shape cache length on all compilable caches

(this diff is the same on all models)